anti-virus-scan.tf

```diff
@@ -194,6 +201,5 @@ resource "aws_lambda_permission" "main_scan" {
   source_account = data.aws_caller_identity.current.account_id
   source_arn     = element(data.aws_s3_bucket.main_scan.*.arn, count.index)
 
-  statement_id = "${var.name_scan}-${element(data.aws_s3_bucket.main_scan.*.id, count.index)}"
+  statement_id = replace("${var.name_scan}-${element(data.aws_s3_bucket.main_scan.*.id, count.index)}", ".", "-")
```

Comment:
Do you have an example of what this is doing so I can review it?

Reply:
using sam example from #11

```hcl
module "s3_anti_virus" {
  source = "trussworks/s3-anti-virus/aws"
  version = "2.1.1"

  av_definition_s3_bucket = "mydefinitionbucket"
  av_scan_buckets               = ["mybucketname.com"]
  lambda_package               = "s3-anti-virus"
  lambda_s3_bucket            = "mylambdabucket"
  lambda_version                  = "v2.0.0"
}
```

```hcl
Terraform v0.12.23
Configuring remote state backend...
Initializing Terraform configuration...
2020/03/27 22:18:26 [DEBUG] Using modified User-Agent: Terraform/0.12.23 TFC/a5cdccec65
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

module.s3_anti_virus.data.aws_caller_identity.current: Refreshing state...
module.s3_anti_virus.data.aws_iam_policy_document.assume_role_scan: Refreshing state...
module.s3_anti_virus.data.aws_region.current: Refreshing state...
module.s3_anti_virus.data.aws_iam_policy_document.assume_role_update: Refreshing state...
module.s3_anti_virus.data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
module.s3_anti_virus.data.aws_s3_bucket.main_scan[0]: Refreshing state...

------------------------------------------------------------------------

Error: "statement_id" doesn't look like a valid statement ID ("^[a-zA-Z0-9-_]+$"): "s3-anti-virus-scan-mybucketname.com"

  on .terraform/modules/s3_anti_virus/anti-virus-scan.tf line 193, in resource "aws_lambda_permission" "main_scan":
 193: resource "aws_lambda_permission" "main_scan" {
```

this PR would set `aws_lambda_permission.main_scan.statement_id = "s3-anti-virus-scan-mybucketname-com"` instead of `"s3-anti-virus-scan-mybucketname.com"`

Reply:
Oh thanks! It's a bit late for me today but I'll try this out Monday and let you know.